File: backend/app/modules/organizations/permissions.py
from app.modules.identity.enums import PermissionKey

ORGANIZATION_PERMISSIONS: dict[PermissionKey, str] = {
    PermissionKey.ORGANIZATION_READ:
        "View organization information.",
        
    PermissionKey.ORGANIZATION_MANAGE:
        "Manage organization settings.",
        
    PermissionKey.ORGANIZATION_MEMBERS_MANAGE:
        "Manage organization members.",
}

File: frontend/src/modules/organizations/permissions.ts
export const ORGANIZATION_PERMISSIONS = {
  ORGANIZATION_READ: "organization.read",
  ORGANIZATION_MANAGE: "organization.manage",
  ORGANIZATION_MEMBERS_MANAGE: "organization.members.manage",
} as const;

export type OrganizationPermission =
  (typeof ORGANIZATION_PERMISSIONS)[keyof typeof ORGANIZATION_PERMISSIONS];

export function hasOrganizationPermission(
  permissions: string[] | undefined,
  permission: OrganizationPermission,
): boolean {
  return permissions?.includes(permission) ?? false;
}